cmd/agent/install_script.sh

```diff
@@ -220,7 +221,26 @@ elif [ "$OS" = "Debian" ]; then
     fi
     printf "\033[34m\n* Installing APT package sources for Datadog\n\033[0m\n"
     $sudo_cmd sh -c "echo 'deb https://${apt_url}/ ${apt_repo_version}' > /etc/apt/sources.list.d/datadog.list"
-    $sudo_cmd apt-key adv --recv-keys --keyserver "${keyserver}" A2923DFF56EDA6E76E55E492D3A80E30382E94DE
+    retries=0
+    while [ "$retries" -le 4 ]; do
+      $sudo_cmd apt-key adv --recv-keys --keyserver "${keyserver}" A2923DFF56EDA6E76E55E492D3A80E30382E94DE && break
+      if [ "$retries" -eq 4 ]; then
+        ERROR_MESSAGE="ERROR
+Couldn't fetch Datadog public key.
+This might be due to a connectivity error with the key server
+or a temporary service interruption.
+*****
+"
+        false
+      fi
+      printf "\033[31m\napt-key failed to retrieve Datadog's public key, retrying in 5 seconds...\n\033[0m\n"
+      sleep 5
+      if [ "$retries" -eq 1 ]; then
+        printf "\033[31mSwitching to backup keyserver\n\033[0m\n"
```

Comment:
Nit: I'd make this message either yellow (`\033[33m`) or blue (`\033[34m`) as it's more of an informative message about what the script is doing.